src/metax_api/api/rest/base/views/file_view.py

```diff
@@ -166,18 +166,36 @@ def partial_update_bulk(self, request, *args, **kwargs):
     @action(detail=False, methods=['post'], url_path="datasets")
     def datasets(self, request):
         """
-        Find out which datasets a list of files belongs to, and return their
+        keys=files: Find out which datasets a list of files belongs to, and return their
         metadata_version_identifiers as a list.
 
+        keys=datasets: Find out which files belong to a list of datasets, and return their
+        identifiers as a list.
+
+        keysonly: return the same list as input, but only with files that belong to some dataset /
+        only with datasets that have files
+
         The method is invoked using POST, because there are limits to length of query
         parameters in GET. Also, some clients forcibly shove parameters in body in GET
         requests to query parameters, so using POST instead is more guaranteed to work.
         """
 
-        if CommonService.get_boolean_query_param(request, 'detailed'):
-            return FileService.get_detailed_datasets_where_file_belongs_to(request.data)
+        keysonly = CommonService.get_boolean_query_param(self.request, 'keysonly')
```

Comment:
self.request -> request as it comes as parameter

Reply:
True, fixed it...